### ISSUE TITLE
Why do I see 2 connections on server even if my conection type is WebSockets

### ISSUE THREAD

**Issue by NamanJain2050** [miguelgrinberg/Flask-SocketIO#1346]
**Your question**
I see two connections on server from my client IP even when my socket connection type is Websockets:

<img width="1312" alt="Screenshot 2020-08-01 at 5 27 44 PM" src="https://user-images.githubusercontent.com/17691278/89101345-aa539a80-d41c-11ea-82e0-1fc41a3b7141.png">
<img width="1678" alt="Screenshot 2020-08-01 at 5 27 27 PM" src="https://user-images.githubusercontent.com/17691278/89101349-b0e21200-d41c-11ea-974a-17d4e7866f8e.png">

Why is this the case ?

**Comment by miguelgrinberg** [comment]
What do you mean by "my socket connection type is Websockets"? What have you done in the server and the client to ensure only WebSocket is used?

**Comment by NamanJain2050** [comment]
I relied on the logs suggesting that the connection have been upgraded to Websockets ? 
1. Is is not the case that the connection type is websockets even if the logs say so ?
2. Is something else has to be done to ensure websocket connection?

**Comment by miguelgrinberg** [comment]
You may not be aware of how the Socket.IO protocol works. The connection is initiated as long-polling, and only once that connection is established the client attempts to "upgrade" to WebSocket. Unless you tell your client to connect directly via WebSocket, it will always initiate with long-polling since that is the default for the protocol.

**Comment by NamanJain2050** [comment]
Yes I understand that the default connection type is long-polling and than only it is upgraded to websocket. 
What I'm really wondering is once we see the log: "upgrade to websocket successful" we can assume that the connection type is now websocket, right ?
And if once the connection type is websocket, we shouldn't see two TCP connections from the client IP. 
Or am I missing something here?

**Comment by miguelgrinberg** [comment]
@NamanJain2050 I'm really not sure how to answer your questions. The server did not issue any connections to anyone, you are asking me why your client made two connections, and I don't even know which client you are using. I suggest you start debugging this from the client, and then if you think the server is doing something wrong you let me know.

**Comment by NamanJain2050** [comment]
I'm using Android Client and socket-io-java-client implementation. I looked more into the client side defaults and found that the client was actually making long-polling connection only and not a websocket one. I changed the defaults to make a websocket connection and now I can only see one connection on server. 
The "upgrade to websocket" log and still seeing two connections on server confused me.
